File: src/main/java/xyz/aerii/athen/mixin/accessors/AbstractContainerScreenAccessor.java
package xyz.aerii.athen.mixin.accessors;

import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
import net.minecraft.world.inventory.Slot;
import org.jetbrains.annotations.Nullable;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

@Mixin(AbstractContainerScreen.class)
public interface AbstractContainerScreenAccessor {
    @Accessor("hoveredSlot")
    @Nullable
    Slot hovered();

    @Accessor("leftPos")
    int leftPos();

    @Accessor("topPos")
    int topPos();
}

File: src/main/kotlin/xyz/aerii/athen/accessors/AbstractContainerScreenAccessor.kt
package xyz.aerii.athen.accessors

import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen
import net.minecraft.world.inventory.Slot
import xyz.aerii.athen.mixin.accessors.AbstractContainerScreenAccessor

val AbstractContainerScreen<*>.hovered: Slot?
    get() = (this as AbstractContainerScreenAccessor).hovered()

val AbstractContainerScreen<*>.left: Int
    get() = (this as AbstractContainerScreenAccessor).leftPos()

val AbstractContainerScreen<*>.top: Int
    get() = (this as AbstractContainerScreenAccessor).topPos()